component-monitor/src/main.rs

```diff
@@ -0,0 +1,238 @@
+            return;
+        }
+    };
+
+    let sentinel_tgt = target.join(sentinel);
+    if sentinel_data == fs::read(&sentinel_tgt).unwrap_or_default() {
+        info!(target: "files", "found current sentinel, skipping");
+        return;
+    }
+
+    cleanup(target, sentinel);
+    info!(target: "files", "copying from {source:?} to {target:?}");
+    for entry in WalkDir::new(&source) {
+        match entry {
+            Ok(e) => {
+                let path = e.path();
+                let relative_path = path.strip_prefix(source).unwrap();
+                if path == source {
+                    continue;
+                }
+                if *(relative_path.to_str().unwrap()) == **sentinel {
+                    continue;
+                }
+
+                let target_path = target.join(relative_path);
+                if path.is_dir() {
+                    match fs::create_dir(&target_path) {
+                        Ok(()) => {
+                            debug!(target: "files", "created {0:?}", &target_path)
+                        }
+                        Err(e) => {
+                            error!(target: "files", "failed to create {0:?}: {e}", &target_path)
+                        }
+                    }
+                } else {
+                    match fs::copy(path, &target_path) {
+                        Ok(_) => {
+                            debug!(target: "files", "copied {0:?} to {1:?}", &path, &target_path);
+                        }
+                        Err(e) => {
+                            error!(target: "files", "failed to copy {0:?} to {1:?}: {e}", &path, &target_path);
+                        }
+                    }
+                }
+            }
+            Err(e) => {
+                error!(target: "files", "{e}");
+            }
+        }
+    }
+
+    match fs::copy(&sentinel_src, &sentinel_tgt) {
+        Ok(_) => {
+            debug!(target: "files", "copied sentinel ({0:?} to {0:?})", sentinel_src);
+        }
+        Err(e) => {
+            error!(target: "files", "failed to copy sentinel: {e}");
+        }
+    }
+}
+
+/// Handle inotify events:
+/// - run cleanup on sentinel deletion
+/// - populate on sentinel written
+/// - panic on source remove
+fn monitor(inotify: &Inotify, args: &Cli) {
+    info!(target: "inotify", "starting event monitoring on {0:?}", &args.source);
+    loop {
+        let events = inotify
+            .read_events()
+            .expect("ERROR: Failed to read inotify events");
+
+        for event in events {
+            debug!(
+                target: "inotify",
+                "handling {0:?} event for {1:?}",
+                event.mask,
+                event.name
+            );
+            match event {
+                InotifyEvent {
+                    name: Some(name),
+                    mask,
+                    ..
+                } if *name == *args.sentinel => match mask {
```

Comment:
(from @TheSignPainter98 and @sminez)
Since `InotifyEvent` is not an enum but a struct, using `match` here is unexpected. It's better to extract the values and match them separately:

```rust
let (event_name, mask) = match event { InotifyEvent { name, mask .. } => (name, mask),
    _ => panic!("..."),
};
```

Moreover, since it seems that you need only the fields from this struct, you can simplify it the following way:

```rust
for InotifyEvent { name, mask, .. } in events {
    ...
}
```